// @scfhq/foundation-uax/src/components/SCFDataGrid/SCFGridToolbarExport.tsx
import * as React from 'react';
import PropTypes from 'prop-types';
import { ButtonProps } from '@mui/material/Button';
import MenuItem from '@mui/material/MenuItem';
import { useGridApiContext } from '@mui/x-data-grid/hooks/utils/useGridApiContext';
import { GridToolbarExportContainer } from '@mui/x-data-grid/components/toolbar/GridToolbarExportContainer';
import { GridCsvExportOptions } from '@mui/x-data-grid/models/gridExport';

interface GridExportDisplayOptions {
  /**
   * If `true`, this export option will be removed from the GridToolbarExport menu.
   * @default false
   */
  disableToolbarButton?: boolean;
}

export interface GridExportMenuItemProps<Options extends {}> {
  hideMenu?: () => void;
  options?: Options & GridExportDisplayOptions;
}

export type GridCsvExportMenuItemProps = GridExportMenuItemProps<GridCsvExportOptions>;

export interface GridToolbarExportProps extends ButtonProps {
  csvOptions?: GridCsvExportOptions & GridExportDisplayOptions;
}

export const GridCsvExportMenuItem = (props: GridCsvExportMenuItemProps) => {
  const apiRef = useGridApiContext();
  const { hideMenu, options } = props;

  if (options?.disableToolbarButton) {
    return null;
  }
  return (
    <MenuItem
      onClick={() => {
        apiRef.current.exportDataAsCsv(options);
        hideMenu?.();
      }}
    >
      {apiRef.current.getLocaleText('toolbarExportCSV')}
    </MenuItem>
  );
};


const SCFGridToolbarExport = React.forwardRef<HTMLButtonElement, GridToolbarExportProps>(
  function GridToolbarExport(props, ref) {
    const { csvOptions = {}, ...other } = props;

    if (csvOptions?.disableToolbarButton) {
      return null;
    }

    return (
      <GridToolbarExportContainer {...other} ref={ref}>
        <GridCsvExportMenuItem options={csvOptions} />
      </GridToolbarExportContainer>
    );
  },
);

SCFGridToolbarExport.propTypes = {
  csvOptions: PropTypes.object,
} as any;

export default SCFGridToolbarExport;
