// @scfhq/foundation-uax/dist/cjs/demos/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCFTreeViewDemo = exports.SCFTooltipDemo = exports.SCFToggleButtonDemo = exports.SCFTextareaAutosizeDemo = exports.SCFTemporaryDrawerDemo = exports.SCFTabDemo = exports.SCFSwitchDemo = exports.SCFSpinnerDemo = exports.SCFSnackbarDemo = exports.SCFSliderDemo = exports.SCFSkeletonDemo = exports.SCFSidebarDemo = exports.SCFSelectDemo = exports.SCFRadioGroupDemo = exports.SCFRadioDemo = exports.SCFPortalDemo = exports.SCFPopoverDemo = exports.SCFPaperDemo = exports.SCFPasswordDemo = exports.SCFNoSsrDemo = exports.SCFMiniDrawerDemo = exports.SCFLoginDemo = exports.SCFListDemo = exports.SCFLinksDemo = exports.SCFInputDemo = exports.SCFIconButtonDemo = exports.SCFHorizontalStepperDemo = exports.SCFGridToolbarExportDemo = exports.SCFGridToolbarDemo = exports.SCFDoubleStepLoginDemo = exports.SCFDividerDemo = exports.SCFDialogDemo = exports.SCFDatePickerDemo = exports.SCFDataGridDemo = exports.SCFCustomMiniDrawerDemo = exports.SCFClickAwayListenerDemo = exports.SCFChipDemo = exports.SCFCheckboxDemo = exports.SCFChangePasswordFormDemo = exports.SCFButtonDemo = exports.SCFBreadcrumbsDemo = exports.SCFBottomNavigationDemo = exports.SCFBasicTableDemo = exports.SCFBasicPaginationDemo = exports.SCFBasicMenuDemo = exports.SCFBadgeDemo = exports.SCFAvatarDemo = exports.SCFAppBarDemo = exports.SCFAlertDemo = exports.SCFAccordionDemo = void 0;
exports.SCFIconDemo = exports.SCFModalDemo = void 0;
const SCFAlertDemo_1 = __importDefault(require("./SCFAlert/SCFAlertDemo"));
exports.SCFAlertDemo = SCFAlertDemo_1.default;
const SCFButtonDemo_1 = __importDefault(require("./SCFButton/SCFButtonDemo"));
exports.SCFButtonDemo = SCFButtonDemo_1.default;
const SCFDoubleStepLoginDemo_1 = __importDefault(require("./SCFDoubleStepLogin/SCFDoubleStepLoginDemo"));
exports.SCFDoubleStepLoginDemo = SCFDoubleStepLoginDemo_1.default;
const SCFLoginDemo_1 = __importDefault(require("./SCFLogin/SCFLoginDemo"));
exports.SCFLoginDemo = SCFLoginDemo_1.default;
const SCFNoSsrDemo_1 = __importDefault(require("./SCFNoSsr/SCFNoSsrDemo"));
exports.SCFNoSsrDemo = SCFNoSsrDemo_1.default;
const SCFPaperDemo_1 = __importDefault(require("./SCFPaper/SCFPaperDemo"));
exports.SCFPaperDemo = SCFPaperDemo_1.default;
const SCFPopoverDemo_1 = __importDefault(require("./SCFPopover/SCFPopoverDemo"));
exports.SCFPopoverDemo = SCFPopoverDemo_1.default;
const SCFPortalDemo_1 = __importDefault(require("./SCFPortal/SCFPortalDemo"));
exports.SCFPortalDemo = SCFPortalDemo_1.default;
const SCFRadioDemo_1 = __importDefault(require("./SCFRadio/SCFRadioDemo"));
exports.SCFRadioDemo = SCFRadioDemo_1.default;
const SCFRadioGroupDemo_1 = __importDefault(require("./SCFRadioGroup/SCFRadioGroupDemo"));
exports.SCFRadioGroupDemo = SCFRadioGroupDemo_1.default;
const SCFSelectDemo_1 = __importDefault(require("./SCFSelect/SCFSelectDemo"));
exports.SCFSelectDemo = SCFSelectDemo_1.default;
const SCFSkeletonDemo_1 = __importDefault(require("./SCFSkeleton/SCFSkeletonDemo"));
exports.SCFSkeletonDemo = SCFSkeletonDemo_1.default;
const SCFSliderDemo_1 = __importDefault(require("./SCFSlider/SCFSliderDemo"));
exports.SCFSliderDemo = SCFSliderDemo_1.default;
const SCFSnackbarDemo_1 = __importDefault(require("./SCFSnackbar/SCFSnackbarDemo"));
exports.SCFSnackbarDemo = SCFSnackbarDemo_1.default;
const SCFSpinnerDemo_1 = __importDefault(require("./SCFSpinner/SCFSpinnerDemo"));
exports.SCFSpinnerDemo = SCFSpinnerDemo_1.default;
const SCFSwitchDemo_1 = __importDefault(require("./SCFSwitch/SCFSwitchDemo"));
exports.SCFSwitchDemo = SCFSwitchDemo_1.default;
const SCFTemporaryDrawerDemo_1 = __importDefault(require("./SCFTemporaryDrawer/SCFTemporaryDrawerDemo"));
exports.SCFTemporaryDrawerDemo = SCFTemporaryDrawerDemo_1.default;
const SCFTextareaAutosizeDemo_1 = __importDefault(require("./SCFTextareaAutosize/SCFTextareaAutosizeDemo"));
exports.SCFTextareaAutosizeDemo = SCFTextareaAutosizeDemo_1.default;
const SCFToggleButtonDemo_1 = __importDefault(require("./SCFToggleButton/SCFToggleButtonDemo"));
exports.SCFToggleButtonDemo = SCFToggleButtonDemo_1.default;
const SCFTooltipDemo_1 = __importDefault(require("./SCFTooltip/SCFTooltipDemo"));
exports.SCFTooltipDemo = SCFTooltipDemo_1.default;
const SCFTreeViewDemo_1 = __importDefault(require("./SCFTreeView/SCFTreeViewDemo"));
exports.SCFTreeViewDemo = SCFTreeViewDemo_1.default;
const SCFBadgeDemo_1 = __importDefault(require("./SCFBadge/SCFBadgeDemo"));
exports.SCFBadgeDemo = SCFBadgeDemo_1.default;
const SCFAccordionDemo_1 = __importDefault(require("./SCFAccordion/SCFAccordionDemo"));
exports.SCFAccordionDemo = SCFAccordionDemo_1.default;
const SCFAppBarDemo_1 = __importDefault(require("./SCFAppBar/SCFAppBarDemo"));
exports.SCFAppBarDemo = SCFAppBarDemo_1.default;
const SCFAvatarDemo_1 = __importDefault(require("./SCFAvatar/SCFAvatarDemo"));
exports.SCFAvatarDemo = SCFAvatarDemo_1.default;
const SCFBasicPaginationDemo_1 = __importDefault(require("./SCFBasicPagination/SCFBasicPaginationDemo"));
exports.SCFBasicPaginationDemo = SCFBasicPaginationDemo_1.default;
const SCFBottomNavigationDemo_1 = __importDefault(require("./SCFBottomNavigation/SCFBottomNavigationDemo"));
exports.SCFBottomNavigationDemo = SCFBottomNavigationDemo_1.default;
const SCFCheckboxDemo_1 = __importDefault(require("./SCFCheckbox/SCFCheckboxDemo"));
exports.SCFCheckboxDemo = SCFCheckboxDemo_1.default;
const SCFChipDemo_1 = __importDefault(require("./SCFChip/SCFChipDemo"));
exports.SCFChipDemo = SCFChipDemo_1.default;
const SCFClickAwayListenerDemo_1 = __importDefault(require("./SCFClickAwayListener/SCFClickAwayListenerDemo"));
exports.SCFClickAwayListenerDemo = SCFClickAwayListenerDemo_1.default;
const SCFDialogDemo_1 = __importDefault(require("./SCFDialog/SCFDialogDemo"));
exports.SCFDialogDemo = SCFDialogDemo_1.default;
const SCFDividerDemo_1 = __importDefault(require("./SCFDivider/SCFDividerDemo"));
exports.SCFDividerDemo = SCFDividerDemo_1.default;
const SCFLinksDemo_1 = __importDefault(require("./SCFLinks/SCFLinksDemo"));
exports.SCFLinksDemo = SCFLinksDemo_1.default;
const SCFListDemo_1 = __importDefault(require("./SCFList/SCFListDemo"));
exports.SCFListDemo = SCFListDemo_1.default;
const SCFDataGridDemo_1 = __importDefault(require("./SCFDataGrid/SCFDataGridDemo"));
exports.SCFDataGridDemo = SCFDataGridDemo_1.default;
const SCFBasicMenuDemo_1 = __importDefault(require("./SCFBasicMenu/SCFBasicMenuDemo"));
exports.SCFBasicMenuDemo = SCFBasicMenuDemo_1.default;
const SCFHorizontalStepperDemo_1 = __importDefault(require("./SCFHorizontalStepper/SCFHorizontalStepperDemo"));
exports.SCFHorizontalStepperDemo = SCFHorizontalStepperDemo_1.default;
const SCFInputDemo_1 = __importDefault(require("./SCFInput/SCFInputDemo"));
exports.SCFInputDemo = SCFInputDemo_1.default;
const SCFBreadcrumbsDemo_1 = __importDefault(require("./SCFBreadcrumbs/SCFBreadcrumbsDemo"));
exports.SCFBreadcrumbsDemo = SCFBreadcrumbsDemo_1.default;
const SCFChangePasswordFormDemo_1 = __importDefault(require("./SCFChangePasswordForm/SCFChangePasswordFormDemo"));
exports.SCFChangePasswordFormDemo = SCFChangePasswordFormDemo_1.default;
const SCFBasicTableDemo_1 = __importDefault(require("./SCFBasicTable/SCFBasicTableDemo"));
exports.SCFBasicTableDemo = SCFBasicTableDemo_1.default;
const SCFCustomMiniDrawerDemo_1 = __importDefault(require("./SCFCustomMiniDrawer/SCFCustomMiniDrawerDemo"));
exports.SCFCustomMiniDrawerDemo = SCFCustomMiniDrawerDemo_1.default;
const SCFMiniDrawerDemo_1 = __importDefault(require("./SCFMiniDrawer/SCFMiniDrawerDemo"));
exports.SCFMiniDrawerDemo = SCFMiniDrawerDemo_1.default;
const SCFGridToolbarDemo_1 = __importStar(require("./SCFDataGrid/SCFGridToolbarDemo"));
exports.SCFGridToolbarExportDemo = SCFGridToolbarDemo_1.default;
Object.defineProperty(exports, "SCFGridToolbarDemo", { enumerable: true, get: function () { return SCFGridToolbarDemo_1.SCFGridToolbarDemo; } });
const SCFTabDemo_1 = __importDefault(require("./SCFTab/SCFTabDemo"));
exports.SCFTabDemo = SCFTabDemo_1.default;
const SCFModalDemo_1 = __importDefault(require("./SCFModal/SCFModalDemo"));
exports.SCFModalDemo = SCFModalDemo_1.default;
const SCFIconDemo_1 = __importDefault(require("./SCFIcon/SCFIconDemo"));
exports.SCFIconDemo = SCFIconDemo_1.default;
const SCFDatePickerDemo_1 = __importDefault(require("./SCFDatePicker/SCFDatePickerDemo"));
exports.SCFDatePickerDemo = SCFDatePickerDemo_1.default;
const SCFSidebarDemo_1 = __importDefault(require("./SCFSidebar/SCFSidebarDemo"));
exports.SCFSidebarDemo = SCFSidebarDemo_1.default;
const SCFPasswordDemo_1 = __importDefault(require("./SCFPassword/SCFPasswordDemo"));
exports.SCFPasswordDemo = SCFPasswordDemo_1.default;
const SCFIconButtonDemo_1 = __importDefault(require("./SCFIconButton/SCFIconButtonDemo"));
exports.SCFIconButtonDemo = SCFIconButtonDemo_1.default;
