// @scfhq/foundation-uax/dist/esm/demos/SCFModal/SCFModalDemo.js
import React, { useState } from "react";
import Typography from "@mui/material/Typography";
import { SCFModal, SCFDivider, SCFButton } from "../../components";
const styleModal = {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 600,
    bgcolor: "background.paper",
    border: "2px solid #000",
    boxShadow: 24,
    p: 4,
};
const SCFModalDemo = () => {
    const [openModal, setOpenModal] = useState(false);
    const handleClickOpenModal = () => {
        setOpenModal(true);
    };
    const handleCloseModal = () => {
        setOpenModal(false);
    };
    return (React.createElement(React.Fragment, null,
        React.createElement(Typography, { variant: "h1", gutterBottom: true, fontWeight: "bold", sx: { mt: 8 } }, "Modal"),
        React.createElement(SCFDivider, { sx: { mb: 4 } }),
        React.createElement(SCFButton, { variant: "contained", onClick: handleClickOpenModal }, "Open modal"),
        React.createElement(SCFModal, { body: React.createElement(React.Fragment, null,
                React.createElement("div", null,
                    React.createElement("h1", null, "TEST modal")),
                React.createElement("div", { className: "botones-modal-grupos" },
                    React.createElement(SCFButton, { variant: "contained", onClick: () => { } }, "Aceptar"),
                    React.createElement(SCFButton, { variant: "outlined", onClick: () => { } }, "Cancelar"))), sx: styleModal, open: openModal, onClose: handleCloseModal, modalTitle: React.createElement("div", { className: "titulo-modal" }, " Seleccionar"), modalDescription: "Update the current branch" })));
};
export default SCFModalDemo;
