// @scfhq/foundation-uax/dist/cjs/components/SCFSidebar/SCFSidebar.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const react_router_dom_1 = require("react-router-dom");
const styles_1 = require("@mui/styles");
const SCFIcon_1 = __importDefault(require("../SCFIcon/SCFIcon"));
const useStyles = (0, styles_1.makeStyles)(() => ({
    dialogTitle: {
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        margin: "10px"
    },
    dialogMargins: {
        margin: "10px"
    },
    sidebar: {
        position: 'fixed',
        left: '0',
        top: '65px',
        height: '100%',
        width: '78px',
        background: '#EFF6F9',
        padding: '6px 0px 0 0',
        'z-index': '990',
        transition: 'all 0.5s ease',
        '& i': {
            color: '#fff',
            height: '60px',
            'min-width': '50px',
            'font-size': '28px',
            'text-align': 'center',
            'line-height': '60px',
        },
        '& li': {
            position: 'relative',
            margin: '0px 0',
            'list-style': 'none',
            height: '70px',
            '& .tooltip': {
                position: 'absolute',
                top: '-20px',
                left: 'calc(100% + 15px)',
                'z-index': '3',
                background: '#fff',
                'box-shadow': '0 5px 10px rgba(0, 0, 0, 0.3)',
                padding: '6px 12px',
                'border-radius': '4px',
                'font-size': '15px',
                'font-weight': '400',
                opacity: '0',
                'pointer-events': 'none',
                transition: '0s',
            },
            '&:hover .tooltip': {
                opacity: '1',
                'pointer-events': 'auto',
                transition: 'all 0.4s ease',
                top: '50%',
                transform: 'translateY(-50%)',
            },
            '& a': {
                display: 'flex',
                height: '100%',
                width: '100%',
                'border-radius': '0',
                'align-items': 'center',
                'text-decoration': 'none',
                transition: 'all 0.4s ease',
                background: 'transparent',
                'border-left': '6px solid transparent',
                '& .links_name': {
                    color: '#EC0000',
                    'font-size': '15px',
                    'font-weight': '400',
                    'white-space': 'nowrap',
                    opacity: '0',
                    'pointer-events': 'none',
                    transition: '0.4s',
                }
            },
            '& a:hover': {
                background: '#fff',
                '& .links_name': {
                    transition: 'all 0.5s ease',
                    /* color: #11101d; */
                },
                '& i': {
                    transition: 'all 0.5s ease',
                    /*  color: #11101d; */
                }
            }
        }
    },
    nav_list: {
        'margin-top': '10px',
        height: '100%',
        'padding-left': '0',
    },
    open: {
        width: '250px',
        '& li .tooltip': {
            display: 'none',
        }
    },
    links_name_open: {
        opacity: '1 !important',
        'pointer-events': 'auto !important',
        padding: '0 15px',
        'font-size': '18px !important',
    },
    routeSelected: {
        'border-left': '6px solid #444444 !important',
        color: '#444444',
        '& i': {
            color: '#444444 !important'
        },
        '&>.links_name': {
            color: '#444444 !important'
        }
    },
    icon: {
        'margin-left': '10px',
        '& i': {
            color: '#EC0000',
            'font-size': '25px !important'
            /*  margin-left: 20px; */
        }
    }
}));
const SCFSidebar = ({ show = true, open = false, links = [{ to: './', icon: 'fas fa-hand-holding-usd', text: 'test Icon' }], }) => {
    const classes = useStyles();
    const renderLink = (link) => (react_1.default.createElement("li", null,
        react_1.default.createElement(react_router_dom_1.NavLink, { to: link.to, activeClassName: classes.routeSelected },
            link.icon && react_1.default.createElement(SCFIcon_1.default, { className: classes.icon, icon: link.icon }),
            react_1.default.createElement("span", { className: 'links_name ' + (open ? classes.links_name_open : '') }, link.text)),
        react_1.default.createElement("span", { className: "tooltip" }, link.text)));
    return (react_1.default.createElement(react_1.default.Fragment, null, show && (react_1.default.createElement(react_router_dom_1.BrowserRouter, null,
        react_1.default.createElement("div", { className: classes.sidebar + ' ' + (open ? classes.open : '') },
            react_1.default.createElement("ul", { className: classes.nav_list }, links.map((link, index) => {
                return (link.to ? (renderLink(link)) : (react_1.default.createElement("li", { key: index },
                    react_1.default.createElement("span", null,
                        link.icon && react_1.default.createElement(SCFIcon_1.default, { className: classes.icon, icon: link.icon }),
                        react_1.default.createElement("span", { className: "links_name" + ' ' + (open ? classes.links_name_open : '') }, link.text)),
                    react_1.default.createElement("span", { className: "tooltip" }, link.text),
                    link.submenus && (link.submenus.map((submenu, indexSubmenu) => {
                        return (react_1.default.createElement("ul", { key: indexSubmenu }, submenu));
                    })))));
            })))))));
};
exports.default = SCFSidebar;
