// @scfhq/foundation-uax/dist/cjs/demos/SCFBasicTable/SCFBasicTableData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bodyData = exports.headerData = void 0;
exports.headerData = [
    'Id',
    'Name',
    'Phone',
    'Email',
    'IBAN',
    'status',
    'actions'
];
exports.bodyData = [
    {
        id: 1,
        Name: 'Claire',
        Phone: '628392928',
        email: 'glambillion0@yellowpages.com',
        iban: 'FR78 3693 8286 36PD AJJD ',
        status: 'A',
        actions: ['editable', 'deletable']
    },
    {
        id: 2,
        Name: 'Joshia',
        email: 'jtattershall8@skyrock.com',
        Phone: '628392928',
        iban: 'SM79 W812 0027 316W IHHV',
        status: 'A',
        actions: ['editable', 'deletable', 'details']
    },
    {
        id: 3,
        Name: 'Matty',
        email: 'mayliff7@boston.com',
        Phone: '628392928',
        iban: 'PT07 6937 4163 3067 0',
        status: 'A',
        actions: ['editable', 'deletable', 'details']
    },
    {
        id: 4,
        Name: 'Matty',
        email: 'glambillion0@yellowpages.com',
        Phone: '628392928',
        iban: 'BE35 5154 4893 5207',
        status: 'I',
        actions: ['reactive']
    },
    {
        id: 5,
        Name: 'Audie',
        email: 'ajehan6@cbc.ca',
        Phone: '628392928',
        iban: 'AD85 9078 9261 XLUE TRXE',
        status: 'A'
    },
    {
        id: 6,
        Name: 'Mellisent',
        email: 'mmcclinton5@booking.com',
        Phone: '628392928',
        iban: 'EE66 1203 0530 0029 3530',
        status: 'I'
    },
    {
        id: 7,
        Name: 'Ewen',
        Phone: '628392928',
        email: 'eflannigan4@unc.edu',
        iban: 'FR47 8093 4143 27ET C8IB',
        status: 'A'
    },
    {
        id: 8,
        Name: 'Torie',
        Phone: '796396122',
        email: 'aearl3@vk.com',
        iban: 'LT42 9866 9507 5238 4828',
        status: 'A'
    },
    {
        id: 9,
        Name: 'valor 9',
        Phone: '628392928',
        email: 'jwalburn2@soundcloud.com',
        iban: 'AE50 2987 4446 9770 7946',
        status: 'A'
    },
    {
        id: 10,
        Name: 'Anatollo',
        Phone: '628392928',
        email: 'csegar1@liveinternet.ru',
        iban: 'ES31 9476 6987 8343 1861',
        status: 'A'
    },
    {
        id: 11,
        Name: 'A',
        Phone: '628392928',
        email: 'jwalburn2@soundcloud.com',
        iban: 'AE50 2987 4446 9770 7946',
        status: 'A'
    },
    {
        id: 12,
        Name: 'I',
        Phone: '628392928',
        email: 'csegar1@liveinternet.ru',
        iban: 'ES31 9476 6987 8343 1861 5295',
        status: 'I'
    }
];
