// @scfhq/foundation-uax/src/demos/SCFModal/SCFModalDemo.tsx
import React, { useState } from "react";
import Typography from "@mui/material/Typography";

import { SCFModal, SCFDivider, SCFButton } from "../../components";

const styleModal = {
  position: "absolute",
  top: "50%",
  left: "50%",
  transform: "translate(-50%, -50%)",
  width: 600,
  bgcolor: "background.paper",
  border: "2px solid #000",
  boxShadow: 24,
  p: 4,
};

const SCFModalDemo = () => {
  const [openModal, setOpenModal] = useState(false);

  const handleClickOpenModal = () => {
    setOpenModal(true);
  };

  const handleCloseModal = () => {
    setOpenModal(false);
  };

  return (
    <>
      <Typography variant="h1" gutterBottom fontWeight={"bold"} sx={{ mt: 8 }}>
        Modal
      </Typography>
      <SCFDivider sx={{ mb: 4 }} />
      <SCFButton variant="contained" onClick={handleClickOpenModal}>
        Open modal
      </SCFButton>
      <SCFModal
        body={
          <>
            <div>
              <h1>TEST modal</h1>
            </div>
            <div className="botones-modal-grupos">
              <SCFButton variant="contained" onClick={() => {}}>
                Aceptar
              </SCFButton>
              <SCFButton variant="outlined" onClick={() => {}}>
                Cancelar
              </SCFButton>
            </div>
          </>
        }
        sx={styleModal}
        open={openModal}
        onClose={handleCloseModal}
        modalTitle={<div className="titulo-modal"> Seleccionar</div>}
        modalDescription={"Update the current branch"}
      ></SCFModal>
    </>
  );
};

export default SCFModalDemo;
