// @scfhq/foundation-uax/dist/cjs/demos/SCFModal/SCFModalDemo.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const Typography_1 = __importDefault(require("@mui/material/Typography"));
const components_1 = require("../../components");
const styleModal = {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 600,
    bgcolor: "background.paper",
    border: "2px solid #000",
    boxShadow: 24,
    p: 4,
};
const SCFModalDemo = () => {
    const [openModal, setOpenModal] = (0, react_1.useState)(false);
    const handleClickOpenModal = () => {
        setOpenModal(true);
    };
    const handleCloseModal = () => {
        setOpenModal(false);
    };
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(Typography_1.default, { variant: "h1", gutterBottom: true, fontWeight: "bold", sx: { mt: 8 } }, "Modal"),
        react_1.default.createElement(components_1.SCFDivider, { sx: { mb: 4 } }),
        react_1.default.createElement(components_1.SCFButton, { variant: "contained", onClick: handleClickOpenModal }, "Open modal"),
        react_1.default.createElement(components_1.SCFModal, { body: react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement("div", null,
                    react_1.default.createElement("h1", null, "TEST modal")),
                react_1.default.createElement("div", { className: "botones-modal-grupos" },
                    react_1.default.createElement(components_1.SCFButton, { variant: "contained", onClick: () => { } }, "Aceptar"),
                    react_1.default.createElement(components_1.SCFButton, { variant: "outlined", onClick: () => { } }, "Cancelar"))), sx: styleModal, open: openModal, onClose: handleCloseModal, modalTitle: react_1.default.createElement("div", { className: "titulo-modal" }, " Seleccionar"), modalDescription: "Update the current branch" })));
};
exports.default = SCFModalDemo;
