// @scfhq/foundation-uax/src/demos/SCFDataGrid/SCFDataGridDemo.tsx
import React from "react";

import Typography from "@mui/material/Typography";
import { SCFDataGrid, SCFDivider } from "../../components";

const SCFDataGridDemo = () => {
  const columns = [
    { field: "id", headerName: "ID", width: 90 },
    {
      field: "firstName",
      headerName: "First name",
      width: 150,
      editable: true,
    },
    {
      field: "lastName",
      headerName: "Last name",
      width: 150,
      editable: true,
    },
    {
      field: "age",
      headerName: "Age",
      type: "number",
      width: 110,
      editable: true,
    },
    {
      field: "fullName",
      headerName: "Full name",
      description: "This column has a value getter and is not sortable.",
      sortable: false,
      width: 160,
      valueGetter: (params: {
        row: { firstName: any; lastName: any; age: any };
      }) =>
        `${params.row.firstName || ""} ${params.row.lastName || ""} ${
          params.row.age || ""
        } `,
    },
  ];

  const rows = [
    { id: 1, lastName: "Snow", firstName: "Jon", age: 35 },
    { id: 2, lastName: "Lannister", firstName: "Cersei", age: 42 },
    { id: 3, lastName: "Lannister", firstName: "Jaime", age: 45 },
    { id: 4, lastName: "Stark", firstName: "Arya", age: 16 },
    { id: 5, lastName: "Targaryen", firstName: "Daenerys", age: null },
    { id: 6, lastName: "Melisandre", firstName: null, age: 150 },
    { id: 7, lastName: "Clifford", firstName: "Ferrara", age: 44 },
    { id: 8, lastName: "Frances", firstName: "Rossini", age: 36 },
    { id: 9, lastName: "Roxie", firstName: "Harvey", age: 65 },
  ];
  return (
    <>
      <Typography variant="h1" gutterBottom fontWeight={"bold"} sx={{mt: 8}}>
          DataGrid
        </Typography>
        <SCFDivider sx={{mb: 4}}/>
        <Typography variant="h3" gutterBottom fontWeight={"bold"} sx={{mt: 8}}>Localizaci칩n</Typography>
        <p>Se puede traducir los textos del grid pasando la prop <b>lang=[es/en]</b> para traducci칩n autom치tica entre esos dos idiomas
           soportados o pas치ndole un objeto a la propiedad <b>locale</b> con las
           traducciones que queramos aplicar. En este caso, los ids de los atributos de dicho objeto deben ser los
           que <a target="_blank" href="https://github.com/mui/mui-x/blob/HEAD/packages/grid/x-data-grid/src/constants/localeTextConstants.ts">define</a> el componente DataGrid de MUI.
        </p>
        
        <div style={{ height: 400, width: "100%" }}>
          <SCFDataGrid
            rows={rows}
            columns={columns}
            pageSize={5}
            rowsPerPageOptions={[5]}
            checkboxSelection
            disableSelectionOnClick
            locale={{ toolbarExport: 'Exportar datos a Excel' }}
            lang="es"
            customExportAction={() => alert("Custon download action demo")}
          />
        </div>
    </>
  );
};

export default SCFDataGridDemo;
