// @scfhq/foundation-uax/src/components/SCFDatePicker/SCFDatePicker.tsx
import React from "react";
import { TextFieldProps } from "@mui/material";
import SCFInput from "../SCFInput/SCFInput";
import { RegisterOptions, UseFormReturn } from "react-hook-form";

type SCFDatePickerProps = TextFieldProps & {
  /**
   * Controls if the component must be rendered
   */
  show?: boolean;
  max?: number;
  min?: number;
  /**
   * React-hook form. If defined, the component will render a React-hook form controller.
   */
  form?: UseFormReturn;
  /**
   * React-hook-forms rules.
   */
  rules?: Omit<RegisterOptions,"valueAsNumber" | "valueAsDate" | "setValueAs" | "disabled">;
};

const SCFDatePicker = ({
  show = true,
  max = 100,
  min = 2,
  variant = "standard",
  form = undefined,
  rules = undefined,
  ...muiprops
}: SCFDatePickerProps) => {
  return (
    <>
      {show && (
        <SCFInput
          {...muiprops}
          form={form}
          rules={rules}
          type="date"
          InputProps={{
            inputProps: {
              max: max,
              min: min,
            },
          }}
          InputLabelProps={{
            shrink: true,
          }}
        />
      )}
    </>
  );
};

export default SCFDatePicker;
