// @scfhq/foundation-uax/dist/esm/themes/Tier1/tier1.js
import { createTheme } from '@mui/material/styles';
const Tier1 = createTheme({
    components: {
        MuiTypography: {
            defaultProps: {
                variantMapping: {
                    display1: 'h1',
                    display2: 'h2',
                    display3: 'h3',
                    bodyLarge: 'p',
                    bodyMedium: 'p',
                    bodySmall: 'p',
                    bodyXSmall: 'p',
                },
            },
        },
        MuiButton: {
            defaultProps: {
                disableElevation: true
            }
        }
    },
    typography: {
        fontFamily: 'SantanderMicroText, Arial',
        fontSize: 16,
        fontWeightLight: 300,
        fontWeightRegular: 400,
        fontWeightMedium: 500,
        fontWeightBold: 700,
        display1: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '3.75rem',
            fontWeight: 400,
            lineHeight: '72px',
            letterSpacing: 0,
        },
        display2: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '3.375rem',
            fontWeight: 400,
            lineHeight: '64px',
            letterSpacing: 0,
        },
        display3: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '3rem',
            fontWeight: 400,
            lineHeight: '60px',
            letterSpacing: 0,
        },
        h1: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '2.625rem',
            lineHeight: '52px',
            '@media (max-width:375px)': {
                fontSize: '2rem',
                lineHeight: '36px',
            },
            letterSpacing: 0,
        },
        h2: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '2.25rem',
            lineHeight: '44px',
            '@media (max-width:375px)': {
                fontSize: '2rem',
                lineHeight: '36px',
            },
            letterSpacing: 0,
        },
        h3: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '2rem',
            lineHeight: '36px',
            '@media (max-width:375px)': {
                fontSize: '1.75rem',
                lineHeight: '32px',
            },
        },
        h4: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '1.75rem',
            lineHeight: '32px',
            '@media (max-width:375px)': {
                fontSize: '1.5rem',
            },
        },
        h5: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '1.5rem',
            lineHeight: '32px',
            '@media (max-width:375px)': {
                fontSize: '1.25rem',
                lineHeight: '24px',
            },
        },
        h6: {
            fontFamily: 'SantanderHeadline, Arial',
            fontSize: '1.25rem',
            lineHeight: '24px',
            '@media (max-width:375px)': {
                fontSize: '1.125rem',
            },
        },
        bodyLarge: {
            fontFamily: 'SantanderMicroText, Arial',
            fontSize: '1.125rem',
            lineHeight: '24px',
        },
        bodyMedium: {
            fontFamily: 'SantanderMicroText, Arial',
            fontSize: '1em',
            lineHeight: '24px',
        },
        bodySmall: {
            fontFamily: 'SantanderMicroText, Arial',
            fontSize: '0.875rem',
            lineHeight: '18px',
        },
        bodyXSmall: {
            fontFamily: 'SantanderMicroText, Arial',
            fontSize: '0.75rem',
            lineHeight: '16px',
        },
        button: {
            fontSize: '0.875rem',
        },
    },
    palette: {
        primary: {
            light: '#ff5633',
            main: '#ec0000',
            dark: '#b00000',
            contrastText: '#fff',
        },
        secondary: {
            light: '#fff',
            main: '#deedf2',
            dark: '#acbbbf',
            contrastText: '#000',
        },
        success: {
            light: '#F0F8F0',
            main: '#63BA68',
            dark: '#3A8340',
        },
        warning: {
            light: '#FFFAEB',
            main: '#FFCC33',
            dark: '#946FDD',
        },
        info: {
            light: '#EBF0FF',
            main: '#3366FF',
            dark: '#0032E6',
        }
    },
});
/** This is customized version of theme for whole application. To use UNICEF theme add MuiThemeProvider at the top level of your app, it will set the custom styles of unicef down to the component tree.
 *  More info: Material-ui theming
 */
export default Tier1;
