// @scfhq/foundation-uax/dist/cjs/App.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
require("./demo-app.scss");
const react_1 = __importDefault(require("react"));
const Container_1 = __importDefault(require("@mui/material/Container"));
const Typography_1 = __importDefault(require("@mui/material/Typography"));
const Box_1 = __importDefault(require("@mui/material/Box"));
const Link_1 = __importDefault(require("@mui/material/Link"));
const Divider_1 = __importDefault(require("@mui/material/Divider"));
const Grid_1 = __importDefault(require("@mui/material/Grid"));
const List_1 = __importDefault(require("@mui/material/List"));
const Toolbar_1 = __importDefault(require("@mui/material/Toolbar"));
const IconButton_1 = __importDefault(require("@mui/material/IconButton"));
const Menu_1 = __importDefault(require("@mui/icons-material/Menu"));
const ListItemText_1 = __importDefault(require("@mui/material/ListItemText"));
const SCFDivider_1 = __importDefault(require("./components/SCFDivider/SCFDivider"));
const ListItemButton_1 = __importDefault(require("@mui/material/ListItemButton"));
const ListItemIcon_1 = __importDefault(require("@mui/material/ListItemIcon"));
const ArrowRight_1 = __importDefault(require("@mui/icons-material/ArrowRight"));
const styles_1 = require("@mui/material/styles");
const Drawer_1 = __importDefault(require("@mui/material/Drawer"));
const AppBar_1 = __importDefault(require("@mui/material/AppBar"));
const Paper_1 = __importDefault(require("@mui/material/Paper"));
const ChevronLeft_1 = __importDefault(require("@mui/icons-material/ChevronLeft"));
const demos_1 = require("./demos");
function Copyright(props) {
    return (react_1.default.createElement(Typography_1.default, { variant: "body2", color: "text.secondary", align: "center", ...props },
        "Copyright © ",
        react_1.default.createElement(Link_1.default, { color: "inherit", href: "https://www.santanderconsumer.es/" }, "Santander Consumer Finance HQ"),
        " ",
        new Date().getFullYear(),
        "."));
}
function App() {
    return (react_1.default.createElement(Container_1.default, { maxWidth: "lg" },
        react_1.default.createElement(Typography_1.default, { variant: "h1", gutterBottom: true, fontWeight: "bold", sx: { mt: 8 } }, "Login"),
        react_1.default.createElement(SCFDivider_1.default, { sx: { mb: 4 } }),
        react_1.default.createElement(Box_1.default, { sx: { px: 2, py: 8, mb: 2 } },
            react_1.default.createElement(Grid_1.default, { container: true, spacing: 2 },
                react_1.default.createElement(Grid_1.default, { item: true, md: 6, sx: { display: "flex", justifyContent: "center" } },
                    react_1.default.createElement(demos_1.SCFLoginDemo, null)),
                react_1.default.createElement(Grid_1.default, { item: true, md: 6, p: 2, sx: { display: "flex", justifyContent: "center" } },
                    react_1.default.createElement(demos_1.SCFDoubleStepLoginDemo, null)))),
        react_1.default.createElement("div", { id: "sCFButtonDemo" },
            react_1.default.createElement(demos_1.SCFButtonDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFBadgeDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFAccordionDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFAppBarDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFAvatarDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFBasicPaginationDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFAlertDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSliderDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSnackbarDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSpinnerDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSwitchDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFTextareaAutosizeDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSkeletonDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFRadioDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFRadioGroupDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSelectDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFTreeViewDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFPortalDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFPopoverDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFPaperDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFNoSsrDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFPasswordDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFChangePasswordFormDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFBottomNavigationDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFCheckboxDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFChipDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFClickAwayListenerDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFDialogDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFModalDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFDividerDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFLinksDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFListDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFDataGridDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFBasicMenuDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFHorizontalStepperDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFInputDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFBreadcrumbsDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFMiniDrawerDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFBasicTableDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFDatePickerDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFIconButtonDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFTabDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFGridToolbarDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFTooltipDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFToggleButtonDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFIconDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFCustomMiniDrawerDemo, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(demos_1.SCFSidebarDemo, null)),
        react_1.default.createElement("div", { id: "sCFInputDemo" },
            react_1.default.createElement(demos_1.SCFInputDemo, null))));
}
const drawerWidth = 240;
const AppBar = (0, styles_1.styled)(AppBar_1.default, {
    shouldForwardProp: (prop) => prop !== "open",
})(({ theme, open }) => ({
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(["width", "margin"], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
    }),
    ...(open && {
        marginLeft: drawerWidth,
        width: `calc(100% - ${drawerWidth}px)`,
        transition: theme.transitions.create(["width", "margin"], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen,
        }),
    }),
}));
const Drawer = (0, styles_1.styled)(Drawer_1.default, {
    shouldForwardProp: (prop) => prop !== "open",
})(({ theme, open }) => ({
    "& .MuiDrawer-paper": {
        position: "relative",
        whiteSpace: "nowrap",
        width: drawerWidth,
        transition: theme.transitions.create("width", {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen,
        }),
        boxSizing: "border-box",
        ...(!open && {
            overflowX: "hidden",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen,
            }),
            width: theme.spacing(7),
            [theme.breakpoints.up("sm")]: {
                width: theme.spacing(9),
            },
        }),
    },
}));
function DashboardContent() {
    const [open, setOpen] = react_1.default.useState(true);
    const toggleDrawer = () => {
        setOpen(!open);
    };
    return (react_1.default.createElement(Box_1.default, { sx: { display: "flex" } },
        react_1.default.createElement(AppBar, { position: "absolute", open: open },
            react_1.default.createElement(Toolbar_1.default, { sx: {
                    pr: "24px", // keep right padding when drawer closed
                } },
                react_1.default.createElement(IconButton_1.default, { edge: "start", color: "inherit", "aria-label": "open drawer", onClick: toggleDrawer, sx: {
                        marginRight: "36px",
                        ...(open && { display: "none" }),
                    } },
                    react_1.default.createElement(Menu_1.default, null)),
                react_1.default.createElement(Typography_1.default, { component: "h1", variant: "h6", color: "inherit", noWrap: true, sx: { flexGrow: 1 } }, "Santander Consumer Finance HQ - Foundation UAX Demo"))),
        react_1.default.createElement(Drawer, { variant: "permanent", open: open },
            react_1.default.createElement(Toolbar_1.default, { sx: {
                    display: "flex",
                    alignItems: "center",
                    justifyContent: "flex-end",
                    px: [1],
                } },
                react_1.default.createElement(IconButton_1.default, { onClick: toggleDrawer },
                    react_1.default.createElement(ChevronLeft_1.default, null))),
            react_1.default.createElement(Divider_1.default, null),
            react_1.default.createElement("a", { href: "#sCFButtonDemo" }, "test"),
            react_1.default.createElement(List_1.default, { component: "nav" },
                react_1.default.createElement(ListItemButton_1.default, { component: Link_1.default, href: "#sCFButtonDemo" },
                    react_1.default.createElement(ListItemIcon_1.default, null,
                        react_1.default.createElement(ArrowRight_1.default, null)),
                    react_1.default.createElement(ListItemText_1.default, { primary: "SCFButton" })),
                react_1.default.createElement(ListItemButton_1.default, { href: "#sCFInputDemo" },
                    react_1.default.createElement(ListItemIcon_1.default, null,
                        react_1.default.createElement(ArrowRight_1.default, null)),
                    react_1.default.createElement(ListItemText_1.default, { primary: "SCFInput" })))),
        react_1.default.createElement(Box_1.default, { component: "main", sx: {
                backgroundColor: (theme) => theme.palette.mode === "light"
                    ? theme.palette.grey[100]
                    : theme.palette.grey[900],
                flexGrow: 1,
                height: "100vh",
                overflow: "auto",
            } },
            react_1.default.createElement(Toolbar_1.default, null),
            react_1.default.createElement(Container_1.default, { maxWidth: "lg", sx: { mt: 4, mb: 4 } },
                react_1.default.createElement(Grid_1.default, { container: true, spacing: 0 },
                    react_1.default.createElement(Grid_1.default, { item: true, xs: 12 },
                        react_1.default.createElement(Paper_1.default, { sx: { p: 2, display: "flex", flexDirection: "column" } },
                            react_1.default.createElement(App, null)))),
                react_1.default.createElement(Copyright, { sx: { pt: 4 } })))));
}
function Dashboard() {
    return react_1.default.createElement(DashboardContent, null);
}
exports.default = Dashboard;
