// @scfhq/foundation-uax/dist/cjs/stories/SCFDataGrid.stories.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataGrid = void 0;
const react_1 = __importDefault(require("react"));
const components_1 = require("../components");
const x_data_grid_1 = require("@mui/x-data-grid");
// import Toolbar from "@mui/material/Toolbar";
// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
exports.default = {
    title: "Foundation UAX/SCFDataGrid",
    component: components_1.SCFDataGrid,
};
// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => (react_1.default.createElement("div", { style: { height: 400, width: "100%" } },
    react_1.default.createElement(components_1.SCFDataGrid, { ...args })));
const columns = [
    { field: "id", headerName: "ID", width: 90 },
    {
        field: "firstName",
        headerName: "First name",
        width: 150,
        editable: true,
    },
    {
        field: "lastName",
        headerName: "Last name",
        width: 150,
        editable: true,
    },
    {
        field: "age",
        headerName: "Age",
        type: "number",
        width: 110,
        editable: true,
    },
    {
        field: "fullName",
        headerName: "Full name",
        description: "This column has a value getter and is not sortable.",
        sortable: false,
        width: 160,
        valueGetter: (params) => `${params.row.firstName || ""} ${params.row.lastName || ""} ${params.row.age || ""} `,
    },
];
const rows = [
    { id: 1, lastName: "Snow", firstName: "Jon", age: 35 },
    { id: 2, lastName: "Lannister", firstName: "Cersei", age: 42 },
    { id: 3, lastName: "Lannister", firstName: "Jaime", age: 45 },
    { id: 4, lastName: "Stark", firstName: "Arya", age: 16 },
    { id: 5, lastName: "Targaryen", firstName: "Daenerys", age: null },
    { id: 6, lastName: "Melisandre", firstName: null, age: 150 },
    { id: 7, lastName: "Clifford", firstName: "Ferrara", age: 44 },
    { id: 8, lastName: "Frances", firstName: "Rossini", age: 36 },
    { id: 9, lastName: "Roxie", firstName: "Harvey", age: 65 },
];
exports.DataGrid = Template.bind({});
exports.DataGrid.args = {
    show: true,
    rows,
    columns,
    pageSize: 5,
    rowsPerPageOptions: [5],
    checkboxSelection: true,
    disableSelectionOnClick: true,
    components: { Toolbar: x_data_grid_1.GridToolbar },
};
